rest/request.go

```diff
@@ -55,6 +55,12 @@ func (r *Request) BaseUrl() *url.URL {
 		scheme = "http"
 	}
 
+	// HTTP/2.0 gives the default scheme as HTTP even when used with TLS
+	// Check if version 2.0 and TLS is not nil and given back https scheme
+	if scheme == "http" && r.ProtoMajor >= 2 && r.TLS != nil {
```

Comment:
You're right. Even with HTTP/1.1 is behaves the same. Example below

```
$ curl -k https://localhost:3332
```
using `http.ListenAndServeTLS()`
```
2017/02/15 10:35:22 HTTP/1.1                // r.Proto
2017/02/15 10:35:22 1                       // r.ProtoMajor
2017/02/15 10:35:22 false                   // r.TLS == nil
2017/02/15 10:35:40 http                    // r.BaseUrl().Scheme
2017/02/15 10:35:40 http://localhost:3332   // r.BaseUrl()
```